### DISCUSSION TITLE
Adding body to ExtenedObject3D requires the object to be added to the scene first? (Headless mode)

### DISCUSSION THREAD

**Discussion by johnno82** [enable3d/enable3d#393] [Help / Questions] [answered]
Like the title says, would it be a requirement for the body to be created to have the ExtenedObject3D added to the scene first?

Reason I am asking is because In my comoponent based game engine I am adding the physics component that creates the body when the entity is not added to the scene.

Currently using the physics.add.existing() method passing an ExtendedObject3D which contains a 3d model or a mesh as a child.

I am using the "headless" mode, still not sure what it means except I don't pass the THREE.Scene object to my AmmoPhysics object.

Thanks

**Comment by yandeu** [comment, marked as answer]
In headless mode there is no scene. You can only add physics to an Object3D but you can't add it to some sort of visual scene. Once you add physics to an object, it will immediately start interacting with other physical bodies.

Hope this helps :)

**Reply by johnno82** [reply]
It does, thank you!
